### PR TITLE
Enable SCDCalibratePanels for the calibration of panel-type instrument such as TOPAZ and MANDI

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- Original Gitlab Taks:  [SCD289](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/308).
- This is #31325 into `master`.

This PR enables users to use the latest version of `SCDCalibratePanels2` to calibrate TOPAZ and MANDI like instrument.

**To test:**

- use the following code to generate a synthetic data set for TOPAZ
```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
from mantid.geometry import CrystalStructure
import matplotlib.pyplot as plt
import numpy as np

from collections import namedtuple

def convert(dictionary):
    return namedtuple('GenericDict', dictionary.keys())(**dictionary)

# --- setup simulated workspace
lc_natrolite = {
    "a": 18.29,  # A
    "b": 18.64,  # A
    "c": 6.56,  # A
    "alpha": 90,  # deg
    "beta": 90,  # deg
    "gamma": 90,  # deg
}
natrolite = convert(lc_natrolite)

# Generate simulated workspace for TOPAZ
CreateSimulationWorkspace(
    Instrument='TOPAZ',
    BinParams='1,100,10000',
    UnitX='TOF',
    OutputWorkspace='cws',
)
cws = mtd['cws']

# Set the UB matrix for the sample
# u, v is the critical part, we can start with the
# ideal position
SetUB(
    Workspace="cws",
    u='1,0,0',  # vector along k_i, when goniometer is at 0
    v='0,1,0',  # in-plane vector normal to k_i, when goniometer is at 0
    **lc_natrolite,
)


# Generate predicted peak workspace
dspacings = convert({'min': 1.0, 'max': 10.0})
wavelengths = convert({'min': 0.8, 'max': 2.9})

# Collect peaks over a range of omegas
CreatePeaksWorkspace(OutputWorkspace='pws')
omegas = range(0, 180, 3)

for omega in omegas:
    SetGoniometer(
        Workspace="cws",
        Axis0=f"{omega},0,1,0,1",
    )

    PredictPeaks(
        InputWorkspace='cws',
        WavelengthMin=wavelengths.min,
        wavelengthMax=wavelengths.max,
        MinDSpacing=dspacings.min,
        MaxDSpacing=dspacings.max,
        ReflectionCondition='All-face centred',
        OutputWorkspace='_pws',
    )

    CombinePeaksWorkspaces(
        LHSWorkspace="_pws",
        RHSWorkspace="pws",
        OutputWorkspace="pws",
    )

IndexPeaks("pws")
```
- move the instrument around using `MoveInstrumentComponent` and `RotateInstrumentComponent`.
> simple example below
```python
MoveInstrumentComponent(Workspace='pws', ComponentName='moderator', Z=0.002)
```

- use `SCDCalibratePanels` to calibrate `pws`, which should move it to a location close to the answer, the engineering position.

Fixes #xxxx. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Crystal/test/SCDCalibratePanels2Test.h [hunk 1]
```diff
@@ -194,41 +204,73 @@ class SCDCalibratePanels2Test : public CxxTest::TestSuite {
...
+    IAlgorithm_sptr lpf_alg = AlgorithmFactory::Instance().create("LoadParameterFile", 1);
+    lpf_alg->initialize();
+    lpf_alg->setLogging(LOGCHILDALG);
+    lpf_alg->setProperty("Workspace", ws);
+    lpf_alg->setProperty("Filename", xmlFileName);
+    lpf_alg->execute();
+
+    // check translation
+    V3D pos_wrng = pws->getInstrument()->getComponentByName(bankname)->getRelativePos();
+    V3D pos_ref = m_pws->getInstrument()->getComponentByName(bankname)->getRelativePos();
```
Comment (jmborr) [inline Framework/Crystal/test/SCDCalibratePanels2Test.h:240]: Why the relative position? Do `ws` and `m_pws` have the same reference instrument? Why not the absolute position?

File: docs/source/algorithms/SCDCalibratePanels-v2.rst [hunk 2]
```diff
@@ -9,7 +9,7 @@
 Description
 -----------
 
-This calibration algorithm is develped for CORELLI type instrument (pack of tubes).
+This calibration algorithm is develped for TOPAZ type instrument (flat panels).
```
Comment (jmborr) [inline docs/source/algorithms/SCDCalibratePanels-v2.rst:12]: "developed"

File: docs/source/algorithms/SCDCalibratePanels-v2.rst [hunk 3]
```diff
@@ -172,9 +179,10 @@ scientists are working on the following targets:
 1) Fix the current T0 (initial TOF offset) calibration issue for synthetic data where a
    constant zero offset was found regardless of the actual prescirbed T0.
 
-2) The original (version 1) SCDCalibratePanels can also support TOPAZ, which consists of
-   multi-panels banks.
-   It is imporant for the new version of SCDCalibratePanels to provide similar support.
+2) The current data structure (detector representation) is not suitable for calibrating
+   instrument with tube-type detectors (such as CORELLI).
+   Additional work on an imporved internal detector and scattering vector representation
```
Comment (jmborr) [inline docs/source/algorithms/SCDCalibratePanels-v2.rst:184]: "improved"

**Review by jmborr — CHANGES_REQUESTED** [review]

File: Framework/Crystal/test/SCDCalibratePanels2Test.h (hunk 1 above)
Comment (KedoKudo) [reply Framework/Crystal/test/SCDCalibratePanels2Test.h:240]: The RelativePos here means relative to the parent class, not the same meaning as the one in MoveInstrumentComponent.

File: docs/source/algorithms/SCDCalibratePanels-v2.rst (hunk 2 above)
Comment (KedoKudo) [reply docs/source/algorithms/SCDCalibratePanels-v2.rst:12]: typo fixed.

File: docs/source/algorithms/SCDCalibratePanels-v2.rst (hunk 3 above)
Comment (KedoKudo) [reply docs/source/algorithms/SCDCalibratePanels-v2.rst:184]: typo fiexed.

**Comment by jmborr** [comment]
@KedoKudo Can't give the OK until you get those branch conflicts fixed

**Review by jmborr — APPROVED** [review]

**Pull request merged**